saleor/payment/tests/test_payment.py

```diff
@@ -550,3 +562,129 @@ def test_update_payment(gateway_response, payment_txn_captured):
     assert payment.cc_exp_year == gateway_response.payment_method_info.exp_year
     assert payment.cc_exp_month == gateway_response.payment_method_info.exp_month
     assert payment.payment_method_type == gateway_response.payment_method_info.type
+
+
+@freeze_time("2021-06-01 12:00:00")
+@override_settings(UNFINISHED_PAYMENT_TTL=timedelta(days=1))
+@pytest.mark.parametrize(
+    "dates, order_exist, is_active, should_be_released",
+    [
+        (["2021-05-01 12:00:00.0+00:00"], False, True, True),  # release, is to old
+        (
+            ["2021-06-01 12:00:00.0+00:00"],
+            False,
+            True,
+            False,
+        ),  # not release, date is to young
+        (
+            ["2021-05-01 12:00:00.0+00:00", "2021-06-01 12:00:00.0+00:00"],
+            False,
+            True,
+            False,
+        ),  # not release, the second transaction is to young
+        (
+            ["2021-06-01 12:00:00.0+00:00", "2021-05-01 12:00:00.0+00:00"],
+            False,
+            True,
+            False,
+        ),  # not release, the first transaction is to young
+        (
+            ["2021-04-01 12:00:00.0+00:00", "2021-05-01 12:00:00.0+00:00"],
+            False,
+            True,
+            True,
+        ),  # release, both transaction are old
+        (
+            ["2021-05-01 12:00:00.0+00:00"],
+            True,
+            True,
+            False,
+        ),  # not release, order exists
+        (
+            ["2021-05-01 12:00:00.0+00:00"],
+            False,
+            False,
+            False,
+        ),  # not release, is not active
+    ],
+)
+def test_payment_needed_to_release(
+    payment_dummy, dates, order_exist, is_active, should_be_released
+):
+    payment = payment_dummy
+    if not order_exist:
+        payment.order = None
+        payment.save()
+
+    if not is_active:
+        payment.is_active = False
+        payment.save()
+
+    for date in dates:
```

Comment:
Okay, but something like this will not work - https://github.com/spulec/freezegun#auto_tick_seconds-argument
```
@freeze_time("Jan 14th, 2020", auto_tick_seconds=15)
def test_nice_datetime():
    first_time = datetime.datetime.now()
    auto_incremented_time = datetime.datetime.now()
    assert first_time + datetime.timedelta(seconds=15) == auto_incremented_time
```

I think that we can also skip this comment if it is more effort that benefit

Reply:
OK, I've tried different methods, also using generator with freeze_time. It wan't work.